src/System.IO.Pipes/src/System/IO/Pipes/NamedPipeServerStream.Windows.cs

```diff
@@ -39,6 +39,20 @@ public sealed partial class NamedPipeServerStream : PipeStream
                 throw new ArgumentOutOfRangeException(nameof(pipeName), SR.ArgumentOutOfRange_AnonymousReserved);
             }
 
+            PipeSecurity pipeSecurity = null;
+
+            if ((options & PipeOptions.CurrentUserOnly) != 0)
+            {
+                SecurityIdentifier identifier = GetCurrentUser();
+                PipeAccessRule rule = new PipeAccessRule(identifier, PipeAccessRights.ReadWrite, AccessControlType.Allow);
```

Comment:
How is this setting up the security for the given Pipe? I don't see the pipe instance getting passed anywhere.

Reply:
It is being passed through the security attributes, see: https://github.com/dotnet/corefx/pull/26395/files/c50bb469733b32ab61607be7dbe9ae3c41111445#diff-a7e95545fe2b63c35aaee6c4c3de3cffR70